### PR TITLE
go: create2_salt is not required for execution

### PR DESCRIPTION
**Pull request by chfast**
Pulled from #611.

### CODE REVIEW

File: bindings/go/evmc/evmc.go [hunk 1]
```diff
@@ -42,7 +42,7 @@ static struct evmc_result execute_wrapper(struct evmc_vm* vm,
 		input_data,
 		input_size,
 		*value,
-		*create2_salt,
+		{{0}}, // create2_salt: not required for execution
```
Comment (chfast) [inline bindings/go/evmc/evmc.go:45]: I changed that from the original commit because `= {}` is not valid ISO C syntax.

**Review by yperbasis — APPROVED** [review]

File: include/evmc/evmc.h [hunk 2]
```diff
@@ -137,7 +137,7 @@ struct evmc_message
     /**
      * The optional value used in new contract address construction.
      *
-     * Ignored unless kind is EVMC_CREATE2.
+     * Ignored unless kind is EVMC_CREATE2. Not required when invoking evmc_execute_fn().
```
Comment (gumb0) [inline include/evmc/evmc.h:140]: Minor suggestion to reword, but it's up to you.
```suggestion
     * Needed only for host to calculate created address when kind is EVMC_CREATE2. Ignored in evmc_execute_fn().
```

**Review by gumb0 — APPROVED** [review]

File: include/evmc/evmc.h (hunk 2 above)
Comment (chfast) [reply include/evmc/evmc.h]: Thanks.

**Comment by Mohsenghasemi12** [comment]
خوبه

**Pull request merged**